core/src/main/java/org/geysermc/geyser/level/block/BlockStateValues.java

```diff
@@ -458,6 +457,30 @@ public static int getWaterLevel(int state) {
         return WATER_LEVEL.getOrDefault(state, -1);
     }
 
+    /**
+     * Get the height of water from the block state
+     * This is used in FishingHookEntity to create splash sounds when the hook hits the water. In addition,
+     * CollisionManager uses this to determine if the player's eyes are in water.
+     *
+     * @param state BlockState of the block
+     * @return The water height or -1 if the block does not contain water
+     */
+    public static double getWaterHeight(int state) {
+        int waterLevel = BlockStateValues.getWaterLevel(state);
+        if (BlockRegistries.WATERLOGGED.get().contains(state)) {
+            waterLevel = 0;
+        }
+        if (waterLevel >= 0) {
+            double waterHeight = 1 - (waterLevel + 1) / 9.0;
```

Comment:
Same deal with the magic number here.

Reply:
Both magic numbers refer to the number of levels of water. I've added a constant `NUM_WATER_LEVELS` to address this.